runtime/commonMain/src/kotlinx/serialization/internal/CollectionSerializers.kt

```diff
@@ -140,6 +140,45 @@ sealed class MapLikeSerializer<TKey, TVal, TCollection, TBuilder: MutableMap<TKe
     }
 }
 
+abstract class PrimitiveArraySerializer<TElement, TCollection, TBuilder : PrimitiveArraySerializer<TElement, TCollection, TBuilder>.Builder>
```

Comment:
Also: it isn't `TCollection`, it's `TArray`.
Additionally, I would omit `T` as it does not carry any additional information, but affects readability